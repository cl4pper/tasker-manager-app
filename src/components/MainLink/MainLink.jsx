import React from 'react';
import classNames from 'classnames';

// STYLE
import './MainLink.scss';

function MainLink(props) {
	const { to, logo, light, children } = props;
	return (
		<a
			className={classNames('MainLink', {
				'MainLink--dark': !light,
				'MainLink--light': light,
			})}
			href={to}
			target="_blank">
			{logo ? <span className="MainLink__logo">{logo}</span> : null}
			{children}
		</a>
	);
}

export { MainLink };
