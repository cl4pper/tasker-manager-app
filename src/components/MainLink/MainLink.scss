.MainLink {
  @include fontText("md");
  text-decoration: none;
  cursor: pointer;
  display: flex;

  @media (max-width: 500px) {
    @include fontText("lg");
  }

  &__logo {
    color: inherit;
    width: size("lg");
    height: size("lg");
    margin-right: size("xs");
  }

  &--dark {
    color: $text-color-primary;
  }

  &--light {
    color: $text-color-secondary;
  }
}
