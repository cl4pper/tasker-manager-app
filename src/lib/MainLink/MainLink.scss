.MainLink {
	@include fontText("md");
	color: $color-secondary;
}
