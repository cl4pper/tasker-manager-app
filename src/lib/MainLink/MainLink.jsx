import React from 'react';
import PropTypes from 'prop-types';

// STYLE
import './MainLink.scss';

const MainLink = (props) => {
	const { content, href } = props;

	return (
		<a className="MainLink" href={href}>
			{content}
		</a>
	);
};

MainLink.propTypes = {
	content: PropTypes.string.isRequired,
	href: PropTypes.string
};

export { MainLink };
