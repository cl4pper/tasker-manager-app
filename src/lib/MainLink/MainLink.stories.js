import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';
import { MainLink } from './MainLink';

const stories = storiesOf('MainLink', module);
stories.addDecorator(withKnobs);

const props = {
	content: 'Google',
	href: 'google.com'
};

stories.add('Default', () => (
	<MainLink content={text('content', props.content, null)} href={text('href', props.href, null)} />
));
