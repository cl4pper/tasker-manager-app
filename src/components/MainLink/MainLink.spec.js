import React from 'react';
import renderer from 'react-test-renderer';
import { MainLink } from './MainLink';

describe('MainLink', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<MainLink to="" light={false}>Link</MainLink>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
