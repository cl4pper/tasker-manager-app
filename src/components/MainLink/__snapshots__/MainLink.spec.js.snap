// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLink snapshot renders 1`] = `
<a
  className="MainLink MainLink--dark"
  href=""
  target="_blank"
>
  Link
</a>
`;
